hs_core/hydroshare/resource.py

```diff
@@ -722,9 +724,11 @@ def add_resource_files(pk, *files, **kwargs):
         utils.create_empty_contents_directory(resource)
     else:
         if resource.resource_type == "CompositeResource" and auto_aggregate:
-            utils.check_aggregations(resource, ret)
+            new_logical_files = utils.check_aggregations(resource, ret)
+            if len(new_logical_files) > 0:
+                return ret, True
 
-    return ret
+    return ret, False
```

Comment:
@pkdash I don't love this approach of adding a second return value to indicate whether aggregations were generated during file addition, but I didn't think of a slicker solution. Let me know if you have a better idea, thanks?!

Reply:
@devincowan  It seems this is the only place where this new flag is really used:
https://github.com/hydroshare/hydroshare/pull/5109/files#diff-455e3bd5828f0016dfc70fba20d67b20cacf8c426a65e1060b4ffeac1ea6fda4R400

In that case you can get count of aggregations before the file add process here:
https://github.com/hydroshare/hydroshare/pull/5109/files#diff-455e3bd5828f0016dfc70fba20d67b20cacf8c426a65e1060b4ffeac1ea6fda4R356

And then do another aggregation count after the file add process. The difference should tell us if new aggregations created.

Reply:
thanks @pkdash for the great suggestion! I did just that. Ready for review.